vllm/v1/worker/gpu_input_batch.py

```
@@ -356,6 +363,50 @@ def remove_request(self, req_id: str) -> Optional[int]:
             self.allowed_token_ids_mask_cpu_tensor[req_index].fill_(False)
         return req_index
 
+    def swap_states(self, i1: int, i2: int) -> None:
+        old_id_i1 = self._req_ids[i1]
+        old_id_i2 = self._req_ids[i2]
+        self._req_ids[i1], self._req_ids[i2] =\
+            self._req_ids[i2], self._req_ids[i1] # noqa
+        self.req_output_token_ids[i1], self.req_output_token_ids[i2] =\
+            self.req_output_token_ids[i2], self.req_output_token_ids[i1]
+        if old_id_i1 is not None and old_id_i2 is not None:
```

what happens if one is None and not the other?

---

Good catch. I added an assert to ensure we would not run into silent failures.